gateway/buffered-logger.go

```diff
@@ -0,0 +1,72 @@
+package gateway
+
+import (
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// implements a buffered logger mainly to be used in tests
+
+type BufferedLog struct {
```

Comment:
Please move to `internal/log` when it's ready :)

Reply:
done